logictest: add ability to call skip.UnderStress from Test-script

michae2 opened the pull request:
**logictest: add ability to call skip.UnderStress from Test-script**

We already had a `skip` command in our fork of Test-script. Flesh it out
a bit with options for `skip.UnderStress` and family. The new syntax
comes in three forms:

```
skip <ISSUE> [args...]
skip ignorelint [args...]
skip under <deadlock/race/stress/stressrace/metamorphic/duress> [ISSUE] [args...]
```

The first form calls `skip.WithIssue`, the second form calls
`skip.IgnoreLint`, and the third form calls `skip.UnderStress` and
family.

Before this change, the only supported form was `skip [ISSUE]` which
called `skip.IgnoreLint`. With this cange, commands in this original
form will instead call `skip.WithIssue`. AFAIK there are no existing
skip commands currently, so this should be a moot point.

Also fix one mistaken use of `skip` which should have been `skipif`.

Informs: #118627

Epic: None

Release note: None

---

**exec: skip distsql_tenant_locality under duress**

In #118769 we added a skip under duress for the distsql_tenant_locality
logic test due to #118627.

Now that the `skip` command supports calling `skip.UnderDuress()`, move
the skip for distsql_tenant_locality from the logictest template to the
test script itself.

Informs: #118627

Epic: None

Release note: None

michae2 committed:
logictest: add ability to call skip.UnderStress from Test-script

We already had a `skip` command in our fork of Test-script. Flesh it out
a bit with options for `skip.UnderStress` and family. The new syntax
comes in three forms:

```
skip <ISSUE> [args...]
skip ignorelint [args...]
skip under <deadlock/race/stress/stressrace/metamorphic/duress> [ISSUE] [args...]
```

The first form calls `skip.WithIssue`, the second form calls
`skip.IgnoreLint`, and the third form calls `skip.UnderStress` and
family.

Before this change, the only supported form was `skip [ISSUE]` which
called `skip.IgnoreLint`. With this cange, commands in this original
form will instead call `skip.WithIssue`. AFAIK there are no existing
skip commands currently, so this should be a moot point.

Also fix one mistaken use of `skip` which should have been `skipif`.

Informs: #118627

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/118820)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm! just had an optional nit

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-:-Nq30-AJ3BW1x9KHgAu9:b8hl3qz)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @michae2 and @rickystewart)

___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-Nq2zqdX6RbgAy4diS34):*
technically it was getting used (e.g. bf60b9c70e193afa6c55238f899f9f1eb8ce3538), but we have been staying on top of unskipping tests
___
*[`pkg/sql/logictest/logic.go` line 3071 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-Nq3-FKoD1oTGWXdGWqZ:-Nq3-FKoD1oTGWXdGWq_:btnersf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a02a8e5e50720095fa35793db8313bd16f1d722e/pkg/sql/logictest/logic.go#L3071)):*
> ```Go
> 			}
> 
> 			if githubIssueID, args := parse(fields[1:]); githubIssueID != -1 {
> ```

nit: i think we could avoid this separate if and do everything in the switch:
```
switch fields[1] {
case "ignorelint":
  // ...
case "under":
  // ...
case "mysql", "mssql", "postgresql", "cockroachdb", "config", "backup-restore":
  // ...
default:
  githubIssueID, args := parse(fields[1:])
  if githubIssueID < 0 {
    return errors.Errorf("unsupported skip command: %v", fields[1])
  }
  // ...
}
```
___
*[`pkg/sql/logictest/testdata/logic_test/udf_schema_change` line 332 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-Nq3-uKH7EbO-1EoIffL:-Nq3-uKI-pAWvfqSUQd1:b-j92saq) ([raw file](https://github.com/cockroachdb/cockroach/blob/a02a8e5e50720095fa35793db8313bd16f1d722e/pkg/sql/logictest/testdata/logic_test/udf_schema_change#L332)):*
> ```
> ALTER FUNCTION f_test_sc(INT) SET SCHEMA test_alter_sc;
> 
> skipif config local-mixed-23.1
> ```

great catch!


<!-- Sent from Reviewable.io -->

rickystewart commented:
The CI failure seems related, no?

michae2 commented:
> The CI failure seems related, no?

Yes, I looked briefly yesterday but got pulled away. I will fix it.

michae2 committed:
exec: skip distsql_tenant_locality under duress

In #118769 we added a skip under duress for the distsql_tenant_locality
logic test due to #118627.

Now that the `skip` command supports calling `skip.UnderDuress()`, move
the skip for distsql_tenant_locality from the logictest template to the
test script itself.

Informs: #118627

Epic: None

Release note: None

michae2 reviewed (commented):
TFTRs!

I fixed the CI failure. Also added another commit moving the skip from #118769 to use the new command.

Not sure if this needs another look. I'll wait until later today to bors it in case either of you want to look again.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-:-Nq433hV9WR99AIRRqQ_:bt2i38x)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-Nq2zqdX6RbgAy4diS34):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

technically it was getting used (e.g. bf60b9c70e193afa6c55238f899f9f1eb8ce3538), but we have been staying on top of unskipping tests
</blockquote></details>

TIL! Updated the commit message.
___
*[`pkg/sql/logictest/logic.go` line 3071 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-Nq3-FKoD1oTGWXdGWqZ:-Nq4-u_72rh2B5Mqm1wF:bfbq99i) ([raw file](https://github.com/cockroachdb/cockroach/blob/a02a8e5e50720095fa35793db8313bd16f1d722e/pkg/sql/logictest/logic.go#L3071)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: i think we could avoid this separate if and do everything in the switch:
```
switch fields[1] {
case "ignorelint":
  // ...
case "under":
  // ...
case "mysql", "mssql", "postgresql", "cockroachdb", "config", "backup-restore":
  // ...
default:
  githubIssueID, args := parse(fields[1:])
  if githubIssueID < 0 {
    return errors.Errorf("unsupported skip command: %v", fields[1])
  }
  // ...
}
```
</blockquote></details>

Good point, done.
___
*[`pkg/sql/logictest/testdata/logic_test/udf_schema_change` line 332 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118820#-Nq3-uKH7EbO-1EoIffL:-Nq431r-6GsmYWkS_60g:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a02a8e5e50720095fa35793db8313bd16f1d722e/pkg/sql/logictest/testdata/logic_test/udf_schema_change#L332)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

great catch!
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

rickystewart commented:
Thanks. This is a colossal improvement.

rafiss reviewed (approved):
> Reviewed 1 of 3 files at r1, 2 of 2 files at r2, 13 of 13 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118820)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @michae2)



<!-- Sent from Reviewable.io -->
